src/engine_core/rendering/src/Shared/RenderableNode.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <glm/mat4x4.hpp>
+#include <vector>
+#include <memory>
+#include "IRenderable.hpp"
+namespace Hush {
+	/// @brief Common renderable node for scenes with multiple children to render
+	/// the Draw function *MUST* be called (recursed down) for every implementation
+	/// PENDING: Unite this into one renderableNode implementation IF AND ONLY IF this becomes a bottleneck
+	class RenderableNode : public IRenderable {
+
+	public:
+		void RefreshTransform(const glm::mat4& parentMatrix) {
+			this->m_worldTransform = parentMatrix * this->m_localTransform;
```

Comment:
For this, we should look into integrating an ECS library, we could have a system that performs this. That way, we avoid coupling the rendering node (which should only contain data) with updating the transforms, also, the nodes become data-only containers, splitting the functionality in two, allowing more flexibility

Reply:
I agree, this behaviour should not be a part of the rendering itself, eventually only the world transform should be passed as part of a render context and that's it, this is a temporary workaround to having updating transforms and children, but we should absolutely consider integrating ECS soon